lms/djangoapps/teacher_dashboard/views.py

```diff
@@ -23,38 +26,43 @@
 from xmodule.modulestore.django import modulestore
 
 from teacher_dashboard.utils import _send_request, has_teacher_access
+from ccx_keys.locator import CCXLocator
 
 
 log = logging.getLogger(__name__)
 
 
 @login_required
-def dashboard_view(request, course_id):
+@coach_dashboard
+def dashboard_view(request, course, ccx=None):
     """
     Teacher dashboard page.
     """
-    # Course is needed for display others tabs
-    course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
-    with modulestore().bulk_operations(course_key):
-        course = get_course_by_id(course_key)
-
-    return render_to_response('teacher_dashboard/dashboard.html', {'course': course, 'course_id': course_id})
+    # right now, we can only have one ccx per user and course
+    # so, if no ccx is passed in, we can safely redirect to that
```

Comment:
> to that?) what is that?

this solution was copied from CCX.
It means, we cannot simply redirect in case of multiple CCXs per course. I left it just FYI.